Small improvements to nicer CI errors

BD103 opened the pull request:
This refactors a few bits of code, but does not change any logic. :)

I recommend viewing the changes one commit at a time, it's easier to digest what's happening!

BD103 committed:
chore!: delete unused prepare module

BD103 committed:
refactor: re-arrange imports

This bumps all the `mod` declarations to the top, all the `pub use` declarations second, and all the private `use` declarations last.

BD103 committed:
feat: `RustChannel::as_str()`

BD103 committed:
refactor: `run_cargo_command_with_json`

Brezak commented on tools/ci/src/commands/mod.rs:
```diff
@@ -115,15 +124,11 @@ pub fn run_cargo_command_with_json(
-            Err(())
-        }
-    }
+    // Don't just leave the child running in the background.
+    let _ = child.wait();
+
+    JsonCommandOutput::from_cargo_output(command_out, command_name.to_string()).map_err(|err| {
+        // Log error, converting it to an `Err(())`.
+        eprintln!("{err}");
+    })
```
Did you tests this? I'm pretty sure this deadlocks.
1. Cargo writes a bunch of data to it's stdout.
2. The pipe buffer gets filled. All subsequent writes block.
3. We wait for the process to finish.
4. Process is stuck waiting for stdout write to finish.

The old behavior would be:
1. Cargo writes a bunch of data to it's stdout.
2. We consume this data in `from_cargo_output`.
2.a If parsing the data fails we drop the child stdout handle closing it. All subsequent writes will fail.
3. We wait for the child to finish.

BD103 replied:
> Did you tests this? I'm pretty sure this deadlocks.

I didn't test this (probably should have!), and it definitely deadlocks. Let me revert that!

Brezak commented on tools/ci/src/commands/mod.rs:
```diff
@@ -96,16 +101,20 @@ pub fn run_cargo_command_with_json(
         .args(flags)
         .envs(env.iter().copied())
+        // TODO: What's this line doing?
         .stdout(Stdio::piped())
```
We need to read what cargo outputs while it's running.
The childs stdio is configured as such:
- `stdin` is inherited from ours. Any reads cargo attempts will be read from our `stdin`.
- `stderr` is inherited from ours. Any writes will show up in console. Due to how cargo determines if output should be colored we should also get color on this output.
- `stdout` is piped directly to us. Any writes to it go directly to a [handle](https://doc.rust-lang.org/std/process/struct.ChildStdout.html) we can [read](https://doc.rust-lang.org/std/io/trait.Read.html). We handle this data in `JsonCommandOutput::from_cargo_output`

BD103 replied:
Thank you, that makes a lot more sense! (I misread `Stdio` as `Stdin`, and was wondering why we were connecting `stdout` and `stdin` together 😅)

BD103 committed:
chore: remove todo comment

More thorough explanation by Brezak here: https://github.com/Brezak/bevy/pull/2#discussion_r1593698955

BD103 committed:
Revert "refactor: `run_cargo_command_with_json`"

This reverts commit 751dd5fa1d188739fbf0b8da31b3f54a2c2a6e18.
